### PR TITLE
date range preset quick selector

### PR DESCRIPTION
**Pull request by sandikbarr**
implements #122

### CODE REVIEW

File: src/app/ngrx/reducers/filter.reducer.ts [hunk 1]
```diff
@@ -18,7 +18,6 @@ export interface FilterModel {
   podcastSeriesId?: number;
   page?: number;
   standardRange?: string;
-  range?: any[];
```
Comment (sandikbarr) [inline src/app/ngrx/reducers/filter.reducer.ts]: stray range that shouldve been removed with my last review that took out the prev and next buttons

File: src/app/shared/menu/dropdown.css [hunk 2]
```diff
@@ -0,0 +1,59 @@
+.dropdown {
```
Comment (sandikbarr) [inline src/app/shared/menu/dropdown.css]: the custom-date-range-dropdown.component.css will instead be using this css with a few custom settings with ticket #123 when the full pop up is implemented that incorporates this standard range list and intervals along with the date range into an advanced menu

File: src/app/shared/menu/date/standard-date-range.component.ts [hunk 3]
```diff
@@ -1,47 +1,52 @@
...
-
-    if (this.interval === INTERVAL_DAILY || this.interval === INTERVAL_WEEKLY || this.interval === INTERVAL_MONTHLY ||
-      (this.interval === INTERVAL_HOURLY && !isMoreThanXDays(40, beginningOfThreeMonthsUTC(), endOfTodayUTC()))) {
-      this.rangeOptions.push(THREE_MONTHS);
-    }
-
-    if (this.interval === INTERVAL_DAILY || this.interval === INTERVAL_WEEKLY || this.interval === INTERVAL_MONTHLY ||
-      (this.interval === INTERVAL_HOURLY && !isMoreThanXDays(40, beginningOfThisYearUTC(), endOfTodayUTC()))) {
-      this.rangeOptions.push(THIS_YEAR);
+    if (this.interval !== INTERVAL_MONTHLY) {
```
Comment (cavis) [inline src/app/shared/menu/date/standard-date-range.component.ts]: A bit nitpicky, but I find this kind of hard to read.  Could it be reorganized more like a switch statement?  (One block for every possible `this.interval`)

File: src/app/shared/menu/date/standard-date-range.component.ts [hunk 4]
```diff
@@ -1,47 +1,52 @@
 import { Component, Input, Output, EventEmitter, OnChanges } from '@angular/core';
-import { IntervalModel, INTERVAL_HOURLY, INTERVAL_DAILY, INTERVAL_WEEKLY, INTERVAL_MONTHLY } from '../../../ngrx';
-import { isMoreThanXDays, endOfTodayUTC,
-  beginningOfThreeMonthsUTC, beginningOfThisYearUTC,
-  TODAY, THIS_WEEK, TWO_WEEKS, THIS_MONTH, THREE_MONTHS, THIS_YEAR,
-  YESTERDAY, LAST_WEEK, PRIOR_TWO_WEEKS, LAST_MONTH, PRIOR_THREE_MONTHS, LAST_YEAR } from '../../util/date.util';
+import { IntervalModel, INTERVAL_HOURLY, INTERVAL_MONTHLY } from '../../../ngrx';
+import { dayOfWeekDateFormat, dayMonthDateFormat, monthDateYearFormat,
```
Comment (cavis) [inline src/app/shared/menu/date/standard-date-range.component.ts]: So many imports!  Wondering if this is a case where `import * as dateUtils from 'date.util'` would actually clean this up.

File: src/app/shared/menu/date/standard-date-range.component.ts (hunk 3 above)
Comment (cavis) [reply src/app/shared/menu/date/standard-date-range.component.ts]: And yeah, as I continue reading, `getRangeDesc()` is much more intuitive than this one.

File: src/app/shared/util/date.util.spec.ts [hunk 5]
```diff
@@ -1,13 +1,14 @@
...
+  beginningOfThisMonthUTC, beginningOfLastMonthUTC, endOfLastMonthUTC, beginningOfLast28DaysUTC, beginningOfLast30DaysUTC,
+  beginningOfThisMonthPlusTwoMonthsUTC, beginningOfLast90DaysUTC,
+  beginningOfThisYearUTC, beginningOfLast365DaysUTC,
   getBeginEndDateFromStandardRange, getStandardRangeForBeginEndDate, getMillisecondsOfInterval,
   roundDateToBeginOfInterval, roundDateToEndOfInterval, getAmountOfIntervals,
   UTCDateFormat, dayOfWeekDateFormat, dayMonthDateFormat, monthDateYearFormat, monthYearFormat, hourlyDateFormat,
-  TODAY, YESTERDAY, THIS_WEEK, LAST_WEEK, TWO_WEEKS, PRIOR_TWO_WEEKS, THIS_MONTH, LAST_MONTH,
-  THREE_MONTHS, PRIOR_THREE_MONTHS, THIS_YEAR, LAST_YEAR } from './date.util';
+  THIS_WEEK, LAST_WEEK, LAST_7_DAYS, THIS_WEEK_PLUS_7_DAYS, THIS_MONTH, LAST_MONTH, LAST_28_DAYS, LAST_30_DAYS,
+  THIS_MONTH_PLUS_2_MONTHS, LAST_90_DAYS, THIS_YEAR, LAST_365_DAYS, OTHER } from './date.util';
```
Comment (cavis) [inline src/app/shared/util/date.util.spec.ts]: Could also just import * here.

Comment (sandikbarr) [reply src/app/shared/util/date.util.spec.ts]: Not using import * is actually killing me but I guess I kinda thought we were anti import * and sided more with being specific with imports, no unused imports. I'd prefer using barrel files and doing import * in most cases.

Comment (cavis) [reply src/app/shared/util/date.util.spec.ts]: Yeah, if it's just a few things (or angular/ngrx core components), I usually side with specific imports.  But yeah, these huge imports are starting to get to me.  Or Todd Motto is subliminally influencing me.

**Comment by sandikbarr** [comment]
alrighty @cavis back to you

File: src/app/shared/menu/date/standard-date-range.component.ts [hunk 6]
```diff
@@ -28,30 +22,33 @@ export class StandardDateRangeComponent implements OnChanges {
...
+  getRanges() {
+    switch (this.interval) {
+      case INTERVAL_HOURLY:
+        return [
+          [dateUtil.THIS_WEEK, dateUtil.LAST_WEEK, dateUtil.LAST_7_DAYS],
+          [dateUtil.THIS_WEEK_PLUS_7_DAYS],
+          [dateUtil.THIS_MONTH, dateUtil.LAST_MONTH, dateUtil.LAST_28_DAYS, dateUtil.LAST_30_DAYS]
+        ];
+      case INTERVAL_DAILY:
+      case INTERVAL_WEEKLY:
```
Comment (cavis) [inline src/app/shared/menu/date/standard-date-range.component.ts]: Niiiiiice!

**Pull request merged**